crypto/property/property_parse.c

```diff
@@ -453,6 +453,33 @@ int ossl_property_has_optional(const OSSL_PROPERTY_LIST *query)
     return query->has_optional ? 1 : 0;
 }
 
+int ossl_property_is_enabled(OPENSSL_CTX *ctx,  const char *property_name,
+                             const OSSL_PROPERTY_LIST *prop_list)
+{
+    int i;
+    OSSL_PROPERTY_IDX name_id;
+    const PROPERTY_DEFINITION *prop = NULL;
+
+    if (prop_list == NULL)
+        return 0;
+
+    if (!parse_name(ctx, &property_name, 0,&name_id))
+        return 0;
+
+    prop = prop_list->properties;
+    for (i = 0; i < prop_list->n; ++i) {
+        if (prop[i].name_idx == name_id) {
+            return prop[i].optional
+                   || (prop[i].type == PROPERTY_TYPE_STRING
+                       && ((prop[i].oper == PROPERTY_OPER_EQ
+                                && prop[i].v.str_val == ossl_property_true)
+                           || (prop[i].oper == PROPERTY_OPER_NE
+                                && prop[i].v.str_val == ossl_property_false)));
+        }
+    }
+    return 0;
```

Comment:
If the property isn't mentioned in the query, should the return be 1 ?  Above at the [see note below], an optional match (either for or against) is treated as matching.  What's the difference between that case and this?